hermes/bin/src/runtime_extensions/hermes/sqlite/state.rs

```diff
@@ -0,0 +1,108 @@
+// cspell: words mapref
+
+//! Internal state implementation for the `SQLite` module.
+
+use std::collections::HashMap;
+
+use dashmap::{mapref::one::RefMut, DashMap};
+use once_cell::sync::Lazy;
+
+use crate::app::HermesAppName;
+
+/// Represents an individual state for a particular object.
+#[derive(Debug)]
+pub(crate) struct ResourceObjectState {
```

Comment:
This struct should be parametrised by the resource type.
With current implementation it's easy to mess taking an instance of the `db_state` for statement objects and broke an entire state.

Reply:
I can at least parameterized by specifying `<T: ToOwned + PartialEq>`.
So I can use as `ResourceObjectState<usize>`.

But since I tried to parameterized it by `ResourceObjectState<*mut sqlite3>`, it gave me this error...

`*mut sqlite3` cannot be sent between threads safely
within `(u32, *mut sqlite3)`, the trait `Send` is not implemented for `*mut sqlite3`

Reply:
I want a concrete example of the problem you are suggesting we have here @Mr-Leshiy ??

Reply:
Already discussed it in private conversation.